### PR TITLE
chore(gatsby): accept multiple localhost addresses

### LINKED ISSUE #14084: accept multiple url variants for graphiql IDE
## Summary

In usability tests, we discovered that when people type the wrong thing in the browser (e.g. “localhost:8000/_ _ graphiql), we could either redirect to "correct" one or just support multiple different cases. This will help them spend less time looking for the right link.

### Basic example
I'm not sure what the best practice is here. Here are some ideas:

Support various number of underscores before graphiql e.g. _graphiql (1 underscore), __graphiql (2 underscores), ___graphiql (3 underscores), ____graphiql (4 underscores)
Support "graphql" and "graphiql"
Other ideas?

### Motivation

We observed that many users (including ourselves!) say "how many  underscores is this again? I can never remember" and "I always misspell Graphiql. How do you even say Graphiql??" haha.

### PR DESCRIPTION
**Pull request by thomascorthouts**
This is a proposal to match multiple versions of a path for the graphiql endpoint. I also _tried_ to create an e2e test to ensure multiple versions of the pattern get you to the graphiql IDE. I put them in a file where I think they could be, but I would love to get feedback on how to improve this or where it should live inside the code.

## Description

When I saw this issue, I could relate to the struggle of thinking about what the exact endpoint was again to play around in GraphiQL. This PR uses the option of express to use a pattern to match the path, this will match with all endpoints with graphql or graphiql with 1 or more leading underscores(`_`)

## Related Issues

Fixes #14084

### CODE REVIEW

File: packages/gatsby/src/commands/develop.js [hunk 1]
```diff
@@ -125,7 +125,7 @@ async function startServer(program) {
 
   if (process.env.GATSBY_GRAPHQL_IDE === `playground`) {
     app.get(
-      `/___graphql`,
+      `/_+graphi{0,1}ql`,
```
Comment (pieh) [inline packages/gatsby/src/commands/develop.js]: Nice! 

Small nitpick:
What about extracting this to const variable that can be used in both route handlers and add some context in comment before that:
```
/**
 * Pattern matching all endpoints with graphql or graphiql with 1 or more leading underscores
 */
const graphqlEndpoint = `/_+graphi{0,1}ql
```

File: e2e-tests/development-runtime/cypress/integration/functionality/graphql-endpoint.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const endpoints = [`/___graphql`, `/_graphql`, `/___graphiql`]
+
+describe(`The GraphQL endpoint`, () => {
+  endpoints.forEach(endpoint => {
+    it(`Should appear on ${endpoint}`, () => {
+      cy.visit(endpoint)
+      cy.title().should(`eq`, `GraphiQL`)
+    })
+  })
+})
```
Comment (pieh) [inline e2e-tests/development-runtime/cypress/integration/functionality/graphql-endpoint.js]: Awesome!

File: packages/gatsby/src/commands/develop.js (hunk 1 above)
Comment (thomascorthouts) [reply packages/gatsby/src/commands/develop.js]: That sounds like a good idea, thanks!

File: packages/gatsby/src/commands/develop.js [hunk 3]
```diff
@@ -123,9 +123,14 @@ async function startServer(program) {
 
   app.use(cors())
 
+  /**
+   * Pattern matching all endpoints with graphql or graphiql with 1 or more leading underscores
+   */
+  const graphqlEndpoint = `/_+graphi{0,1}ql`
```
Comment (wardpeet) [inline packages/gatsby/src/commands/develop.js]: swap `{0,1}` into `?`. I like this syntax more but I might be the only one :p (so if you don't agree, please ignore)
```suggestion
  const graphqlEndpoint = `/_+graphi?ql`
```

Comment (thomascorthouts) [reply packages/gatsby/src/commands/develop.js]: Good point, I didn't think about using an optional operator instead of a range.

**Comment by DSchau** [comment]
Published as gatsby@2.6.2--again, thank you!

**Pull request merged**